app/src/main/java/com/quarkworks/apartmentgroceries/grocery/AddGroceryItemActivity.java

```diff
@@ -47,52 +55,52 @@ protected void onCreate(final Bundle savedInstanceState) {
          * Set view data
          */
         titleTextView.setText(getString(R.string.title_activity_add_grocery_item));
+
+        /**
+         * Set view OnClickListener
+         */
+        addButton.setOnClickListener(addGroceryButtonOnClick());
+
         setSupportActionBar(toolbar);
         getSupportActionBar().setDisplayShowTitleEnabled(false);
+    }
 
-        addButton.setOnClickListener(new View.OnClickListener() {
+    public View.OnClickListener addGroceryButtonOnClick() {
+        return new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 String groceryItemName = groceryItemNameEditText.getText().toString();
 
                 if (!groceryItemName.isEmpty()) {
-                    SharedPreferences sharedPreferences = getApplication()
-                            .getSharedPreferences(getApplication().getString(R.string.login_or_sign_up_session), 0);
-                    String groupId = sharedPreferences.getString("groupId", null);
-                    String userId = sharedPreferences.getString("userId", null);
+                    String groupId = ((MyApplication)MyApplication.getContext()).getGroupId();
+                    String userId = ((MyApplication)MyApplication.getContext()).getUserId();
 
                     RGroceryItem rGroceryItem = new RGroceryItem();
-                    rGroceryItem.setCreatedBy(sharedPreferences.getString(SyncUser.JsonKeys.USER_ID, null));
-                    rGroceryItem.setGroupId(sharedPreferences.getString(SyncUser.JsonKeys.GROUP_ID, null));
                     rGroceryItem.setName(groceryItemName);
                     rGroceryItem.setGroupId(groupId);
                     rGroceryItem.setCreatedBy(userId);
-                    SyncGroceryItem.add(rGroceryItem)
-                            .setCallbacks(addSuccesCallback, addFailureCallback);
+
+                    Continuation<Boolean, Void> addGroceryItemOnSuccess = new Continuation<Boolean, Void>() {
+                        @Override
+                        public Void then(Task<Boolean> task) throws Exception {
+                            if (task.getResult()) {
+                                HomeActivity.newIntent(AddGroceryItemActivity.this);
+                                Toast.makeText(getApplicationContext(), getString(R.string.add_grocery_item_success),
+                                        Toast.LENGTH_SHORT).show();
+                            } else {
+                                Toast.makeText(getApplicationContext(),
```

Comment:
Another way to make this easier to read it to create variables at the bottom of your class for your callbacks. ex)

    private Continuation<Boolean, Void> addGroceryItemOnSuccess = new Continuation<Boolean, Void>() {
      @Override
      public void then(Task<Boolean> task) throw Exception {
        //do stuff
      }
    };

Then, in your click listener, you all you would have to do is:

    RGroceryItem groceryItem = new RGroceryItem();
    groceryItem.setName(groceryItemName);